pandas/core/indexes/base.py

```diff
@@ -3105,7 +3106,10 @@ def reindex(self, target, method=None, level=None, limit=None, tolerance=None):
         if not isinstance(target, Index) and len(target) == 0:
             attrs = self._get_attributes_dict()
             attrs.pop("freq", None)  # don't preserve freq
-            values = self._data[:0]  # appropriately-dtyped empty array
+            if isinstance(self, ABCRangeIndex):
```

Comment:
What does this need to change for?

Reply:
`self._data[:0]` is an ndarray for RangeIndex, so we can't pass it to self._simple_new below